pkg/pool-manager/pod_pool.go

```diff
@@ -131,11 +137,18 @@ func (p *PoolManager) allocatePodRequestedMac(requestedMac string, pod *corev1.P
 		return err
 	}
 
+	if pod.Name == "" {
+		// we are going to create the podToMacPoolMap in the controller call
+		// because we don't have the pod name in the webhook
+		p.macPoolMap[requestedMac] = false
```

Comment:
why don't you drop it instead? if you are using that true/false as a communication channel between webhook and controller, the attribubute needs to be renamed and properly documented. for me it's really hard to understand this right now

Reply:
added a properly documentation about the variable.

`allocated mac map if the value is false the mac was allocated but the pod is not running yet if it is true there is a pod running. for vm is always true`

This append when the user create a pod with the generateName and not with name. we get the wehook without the pod name and then the kube-controller will create the pod with a name this is why we need also need the allocatemac in the reconcile function for pods